lib/practice_exercises.rb

```diff
@@ -1,13 +1,31 @@
 
-# Time Complexity: ?
-# Space Complexity: ?
+# Time Complexity: O(n)
```

Comment:
Since you are deleting nodes that causes the entire array to shift resulting in O(n<sup>2</sup>) time complexity.